lib/graphql.module.ts

```diff
@@ -147,20 +148,66 @@ export class GraphQLModule implements OnModuleInit {
         this.options,
       );
     }
-    this.apolloServer = new ApolloServer(apolloOptions as any);
-    this.apolloServer.applyMiddleware({
-      app,
-      path,
-      disableHealthCheck,
-      onHealthCheck,
-      cors,
-      bodyParserConfig,
-    });
+
+    if (this.isExpress()) {
+      const { ApolloServer } = loadPackage(
+        'apollo-server-express',
+        '@nestjs/graphql',
+      );
+
+      const apolloServer = new ApolloServer(apolloOptions as any);
+
+      apolloServer.applyMiddleware({
+        app,
+        path,
+        disableHealthCheck,
+        onHealthCheck,
+        cors,
+        bodyParserConfig,
+      });
+
+      this.apolloServer = apolloServer;
+    } else if (this.isFastify()) {
+      const { ApolloServer } = loadPackage(
+        'apollo-server-fastify',
+        '@nestjs/graphql',
+      );
+
+      const apolloServer = new ApolloServer(apolloOptions as any);
+
+      app.register(apolloServer.createHandler());
+
+      this.apolloServer = apolloServer;
+    } else {
+      throw new Error(`no support for current HttpAdapter`);
+    }
 
     if (this.options.installSubscriptionHandlers) {
       this.apolloServer.installSubscriptionHandlers(
         httpAdapter.getHttpServer(),
       );
     }
   }
+
+  private async isExpress(): Promise<Boolean> {
+    try {
+      const { ExpressAdapter } = await import('@nestjs/platform-express');
+
+      return this.httpAdapterHost instanceof ExpressAdapter;
```

Comment:
slightly easier (but the tricky way) is just to compare class names (this.httpAdapterHost.constructor.name)

Reply:
we'll provide better support for this shortly

Reply:
fixed